### PR TITLE
PR: Setting up Angular 2 front end for Leverage

### PR DESCRIPTION
**Pull request by medy-io**
This PR creates the frontend for Leverage app with Angular 2 and gets a basic structure using Angular established.

### CODE REVIEW

**Review by lottspot — APPROVED** [review]
👍

**Comment by medy-io** [comment]
Adding basic Angular structure with common files, mock data, and components.

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "typescript.check.workspaceVersion": false
+}
```
Comment (lottspot) [inline .vscode/settings.json]: You made git sad :(

File: gulp.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+//   return gulp
+//     .src('src/**/*.ts')
+//     .pipe(sourcemaps.init())
+//     .pipe(typescript(tscConfig.compilerOptions))
+//     .pipe(sourcemaps.write('.'))
+//     .pipe(gulp.dest('dist'));
+// });
+
+// gulp.task('build', ['clean', 'copy:libs', 'less', 'compile']);
+gulp.task('default', ['clean', 'less' ]);
```
Comment (lottspot) [inline gulp.js]: #sadgit :'(

File: package.json [hunk 3]
```diff
@@ -1,11 +1,61 @@
...
+    "karma-jasmine": "^1.0.2",
+    "karma-jasmine-html-reporter": "^0.2.2",
+    "protractor": "~4.0.14",
+    "rimraf": "^2.5.4",
+
+    "@types/node": "^6.0.46",
+    "@types/jasmine": "^2.5.36"
+  },
+  "repository": {}
+}
```
Comment (lottspot) [inline package.json]: #stillsadgit :'(

File: src/constants/common/leverage.constants.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class LeverageConstants {
+    public REST_URL: string = '/api/../..';
+}
```
Comment (lottspot) [inline src/constants/common/leverage.constants.ts]: #cryinggit :'(

File: src/homepage.component.routes.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { HomepageComponent } from './homepage/homepage.component';
+
+export const leverageroutes = [
+    {
+        path: '/',
+        component: HomepageComponent,
+    }
+];
```
Comment (lottspot) [inline src/homepage.component.routes.ts]: #unlovedgit :'(

File: src/homepage/homepage.component.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    }
+    
+    ngOnInit() {
+        let r: DataRequest = {request: 'data please'};
+        this.campaigns = this.commonService.getCampaignDataStream(r).campaigns;
+    }
+
+
+
+}
```
Comment (lottspot) [inline src/homepage/homepage.component.ts]: you get the point

**Comment by medy-io** [comment]
A lot of those can get sorted out before the merge. I started with the Angular Quick start template and went from there. Any input would be great and depending on the amount of changes I can have an update by tonight.

**Comment by lottspot** [comment]
It's not a earth shatteringly important thing, I just like keeping git happy. I.e., newlines at the ends of files and no trailing whitespace on lines. Not a blocker IMO, but worth mentioning.

**Comment by medy-io** [comment]
I mean if we aren't strict about it I feel like it can out of hand until we can get some sort of error checking going on, so I'll fix it up.

File: .vscode/settings.json (hunk 1 above)
Comment (medy-io) [reply .vscode/settings.json]: switched to true

File: gulp.js (hunk 2 above)
Comment (medy-io) [reply gulp.js]: un-commented so it now compiles

File: package.json (hunk 3 above)
Comment (medy-io) [reply package.json]: deleted

File: src/constants/common/leverage.constants.ts (hunk 4 above)
Comment (medy-io) [reply src/constants/common/leverage.constants.ts]: deleted so now its just an empty class

File: src/homepage/homepage.component.ts (hunk 6 above)
Comment (medy-io) [reply src/homepage/homepage.component.ts]: spacing closed up and left a space after class

File: src/homepage.component.routes.ts (hunk 5 above)
Comment (medy-io) [reply src/homepage.component.routes.ts]: space added after class

**Pull request merged**